### PR TITLE
Add webpack bundling

### PR DESCRIPTION
**Pull request by NavyAdmiral**
@shanalikhan this is for faster startups. 
influenced by @fabiospampinato

refs https://github.com/shanalikhan/code-settings-sync/issues/656 and https://github.com/shanalikhan/code-settings-sync/issues/673

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -160,12 +161,15 @@
...
         "adm-zip": "^0.4.11",
         "chokidar": "^2.0.2",
         "fs-extra": "^7.0.0",
         "https-proxy-agent": "^2.1.1",
         "lockfile": "^1.0.4",
-        "temp": "^0.8.3"
+        "temp": "^0.8.3",
+        "ts-loader": "^5.2.1",
+        "webpack": "^4.20.2",
+        "webpack-cli": "^3.1.2"
```
Comment (fabiospampinato) [inline package.json]: `ts-loader`, `webpack` and `webpack-cli` should be moved to `devDependencies`

Comment (NavyAdmiral) [reply package.json]: I always forget that :(

**Comment by shanalikhan** [comment]
Can you resolve the conflicts.

**Comment by NavyAdmiral** [comment]
> Can you resolve the conflicts.

Can you just cherry-pick my commits

**Comment by shanalikhan** [comment]
Thanks!

**Comment by shanalikhan** [comment]
Following is my output for creating package in  `vsce package`

```
shan@shan-HP-ZBook-15:~/Documents/code-settings-sync$ vsce package
Executing prepublish script 'npm run vscode:prepublish'...

> code-settings-sync@3.2.0 vscode:prepublish /home/shan/Documents/code-settings-sync
> npm run tslint-check && npm run compile


> code-settings-sync@3.2.0 tslint-check /home/shan/Documents/code-settings-sync
> tslint -c ./tslint.json ./src/**/*.ts ./src/*.ts


> code-settings-sync@3.2.0 compile /home/shan/Documents/code-settings-sync
> webpack --mode production

Hash: 5d2d274795858771c4db
Version: webpack 4.20.2
Time: 6172ms
Built at: 10/15/2018 2:12:32 PM
           Asset     Size  Chunks             Chunk Names
    extension.js  591 KiB       0  [emitted]  main
extension.js.map    2 MiB       0  [emitted]  main
Entrypoint main = extension.js extension.js.map
  [0] external "path" 42 bytes {0} [built]
 [24] external "vscode" 42 bytes {0} [built]
 [46] ./src/localize.ts 4.13 KiB {0} [built]
 [59] ./src/environmentPath.ts 6.29 KiB {0} [built]
 [84] ./src/enums.ts 702 bytes {0} [built]
[119] ./src sync 160 bytes {0} [built]
[161] ./src/lockfile.ts 586 bytes {0} [built]
[171] ./src/service/fileService.ts 7.8 KiB {0} [built]
[172] ./src/setting.ts 2.1 KiB {0} [built]
[214] ./src/extension.ts 1.58 KiB {0} [built]
[236] ./src/sync.ts 45 KiB {0} [built]
[237] ./src/commons.ts 25.9 KiB {0} [built]
[387] ./src/service/githubService.ts 5.2 KiB {0} [built]
[542] ./src/service/pluginService.ts 17.2 KiB {0} [built]
[543] ./src/pragmaUtil.ts 8.21 KiB {0} [built]
    + 529 hidden modules

WARNING in ./src/localize.ts 89:40-89
Critical dependency: the request of a dependency is an expression
 @ ./src/extension.ts

WARNING in ./src/localize.ts 94:18-43
Critical dependency: the request of a dependency is an expression
 @ ./src/extension.ts

WARNING in ./node_modules/node-fetch/lib/index.mjs
Module not found: Error: Can't resolve 'encoding' in '/home/shan/Documents/code-settings-sync/node_modules/node-fetch/lib'
 @ ./node_modules/node-fetch/lib/index.mjs
 @ ./node_modules/@octokit/rest/lib/request/request.js
 @ ./node_modules/@octokit/rest/lib/request/index.js
 @ ./node_modules/@octokit/rest/index.js
 @ ./src/service/githubService.ts
 @ ./src/sync.ts
 @ ./src/extension.ts

WARNING in ./node_modules/adm-zip/util/fileSystem.js
Module not found: Error: Can't resolve 'original-fs' in '/home/shan/Documents/code-settings-sync/node_modules/adm-zip/util'
 @ ./node_modules/adm-zip/util/fileSystem.js
 @ ./node_modules/adm-zip/util/index.js
 @ ./node_modules/adm-zip/adm-zip.js
 @ ./src/util.ts
 @ ./src/commons.ts
 @ ./src/sync.ts
 @ ./src/extension.ts
Created: /home/shan/Documents/code-settings-sync/code-settings-sync-3.2.0.vsix (2092 files, 2.28MB)
```

**Comment by shanalikhan** [comment]
Can you fix it.
Try running the commands
```
vsce package
code --install-extension code-settings-sync-3.2.0.vsix
```

Sync localization doesnt work.

**Pull request merged**